### PR TITLE
add q-connection v0.6.1 with npm auto-update

### LINKED ISSUE #7870: [Request] Add q-connection
**Library name:** q-connection
**Git repository url:** https://github.com/kriskowal/q-connection
**npm package url(optional):**  https://www.npmjs.com/package/q-connection
**License(s):** MIT
**Official homepage:**

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #7870 
@pvnr0082t please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/kriskowal/q-connection
- Official website (optional, not the repository):
- NPM package url (optional): https://www.npmjs.com/package/q-connection
- GitHub / Bitbucket popularity (required):
  - Count of stars: 211
  - Count of watchers: 20
  - Count of forks: 27
- NPM download stats (optional):
  - Downloads in the last day: 0
  - Downloads in the last week: 20
  - Downloads in the last month: 372
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR
- [x] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [x] More than 500 downloads stats per month on npm registry
- [ ] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

close #7870, cc @kriskowal

### CODE REVIEW

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

File: ajax/libs/q-connection/0.6.1/q-connection.js [hunk 1]
```diff
@@ -0,0 +1,342 @@
+var Q = require("q");
+var LruMap = require("collections/lru-map");
+var Map = require("collections/map");
+var UUID = require("./lib/uuid");
+var adapt = require("./adapt");
```
Comment (PeterDaveHello) [inline ajax/libs/q-connection/0.6.1/q-connection.js]: Does this work in browser directly?

Comment (kennynaoh) [inline ajax/libs/q-connection/0.6.1/q-connection.js]: How do I confirm this?
Thanks!

Comment (PeterDaveHello) [inline ajax/libs/q-connection/0.6.1/q-connection.js]: There are many ways to confirm.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by sufuf3** [comment]
@PeterDaveHello I don't think this lib can use in web front-end.
Can we close this PR?
Thank you.

**Comment by PeterDaveHello** [comment]
Let's close it, thanks.

**Pull request closed without merging**